src/modules/webpageUtils.jsm

```diff
@@ -505,6 +505,48 @@ var webpageUtils = {
         printSettings.paperSizeUnit           = Ci.nsIPrintSettings.kPaperSizeMillimeters;
         printSettings.scaling                 = options.ratio;
 
+        { // handling webpage.paperSize
+            let currentDPI = 96; // TODO: where to find it properly?
+            let stringToMillimeters = function(s) {
+                let units = {
+                    'mm': 1.0,
+                    'cm': 10.0,
+                    'in': 25.4,
+                    'px': 25.4 / currentDPI
+                };
+                let endsWith = function(str, suffix) {
```

Comment:
Note that Gecko supports [String.prototype.endsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith) ;-)